server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -67,8 +69,17 @@ class LivyServer extends Logging {
     // Initialize YarnClient ASAP to save time.
     if (livyConf.isRunningOnYarn()) {
       Future { SparkYarnApp.yarnClient }
+    } else {
+      // If recovery is turned on but we are not running on YARN, quit.
+      require(livyConf.get(LivyConf.RECOVERY_MODE) == SessionRecoveryMode.OFF,
```

Comment:
Since we have a warning, I think it should be enough, otherwise it will be failed. I'm inclined not to fail the application since this is not a critical issue.